src/jit/codegenarm64.cpp

```diff
@@ -2446,46 +2450,49 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     //     (MinInt / -1) => ArithmeticException
                     //
                     bool checkDividend = true;
-                    // Do we have a contained immediate for the 'divisorOp'?
-                    if (divisorOp->isContainedIntOrIImmed())
+
+                    // Do we have an immediate for the 'divisorOp'?
+                    // 
+                    if (divisorOp->IsCnsIntOrI())
                     {
-                        GenTreeIntConCommon* intConst = divisorOp->AsIntConCommon();
-                        assert(intConst->IconValue() != 0);      // already checked above by IsZero()
-                        if (intConst->IconValue() != -1)
+                        GenTreeIntConCommon* intConstTree = divisorOp->AsIntConCommon();
+                        ssize_t intConstValue = intConstTree->IconValue();
+                        assert(intConstValue != 0);      // already checked above by IsZero()
+                        if (intConstValue != -1)
                         {                            
                             checkDividend = false;    // We statically know that the dividend is not -1
                         }
                     }
-                    else
+                    else  // insert check for divison by zero 
                     {   
                         // Check if the divisor is zero throw a DivideByZeroException
-                        emit->emitIns_R_I(INS_cmp, cmpSize, divisorReg, 0);
+                        emit->emitIns_R_I(INS_cmp, size, divisorReg, 0);
                         emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
                         genJumpToThrowHlpBlk(jmpEqual, SCK_DIV_BY_ZERO);
-
+                    }
+                    
+                    if (checkDividend)
+                    {
```

Comment:
Just for my understanding - Arm32 legacy jit doesn't seem to be doing div-by-zero and minint-div-by-minus1 checks.   Is temporarily required for arm64 till VM has support for propagating hardware exceptions as managed exceptions?